### PR TITLE
Features/antonyk database 

### PR DESCRIPTION
**Pull request by anantonyk**

### CODE REVIEW

File: src/databaseaccessor/accessor.cpp [hunk 1]
```diff
@@ -1,6 +1,6 @@
 #include "accessor.h"
 #include <QDebug>
-
+#include <QSqlError>
```
Comment (PeterLysyk) [inline src/databaseaccessor/accessor.cpp:3]: separate includes and methods' realizations

File: src/databaseaccessor/accessor.cpp [hunk 2]
```diff
@@ -18,6 +18,6 @@ void Accessor::execQuery(QString queryStr)
    query.prepare(queryStr);
    if (!query.exec())
       {
-           qDebug()<<"not executed query";
+           qDebug() << query.lastError().databaseText();
```
Comment (PeterLysyk) [inline src/databaseaccessor/accessor.cpp:22]: come on)

File: src/filemanager/filemanager.cpp [hunk 3]
```diff
@@ -91,3 +90,10 @@ bool FileManager::sourceFileByTheSameNameExists(const QString &headerName)
 {
     return (QFileInfo::exists(removeExtension(headerName,2).append(".cpp")));
 }
+
+QString FileManager::projectFileName =QString();
```
Comment (PeterLysyk) [inline src/filemanager/filemanager.cpp:94]: `QString FileManager::projectFileName = QString();`

File: src/filemanager/filemanager.h [hunk 4]
```diff
@@ -15,6 +15,10 @@ class FileManager
     void createProjectFile(const QString &path);
     bool projectExists(const QString &path);
     bool sourceFileByTheSameNameExists(const QString &headerName);
+    static QString getProjectFileName();
+    private:
+        static QString projectFileName;
```
Comment (PeterLysyk) [inline src/filemanager/filemanager.h:20]: `m`PprojectFileName

File: src/mainwindow/mainwindow.cpp [hunk 5]
```diff
@@ -795,11 +760,6 @@ void MainWindow::onAboutTriggered()
...
 }
 
-void MainWindow::onReferenceTriggered()
-{
-    // Reference Assistant module has been cut off, so just idle
-}
-
 void MainWindow::onUserGuideTriggered()
 {
     //
```
Comment (PeterLysyk) [inline src/mainwindow/mainwindow.cpp:795]: what's that?

File: src/mainwindow/mainwindow.cpp [hunk 6]
```diff
@@ -1064,3 +1011,44 @@ void MainWindow::databaseDisconnect()
 {
     delete db;
 }
+
+void MainWindow::restoreDatabaseFile()
+{
+    setDatabaseFileName();
+    QString projectFileName = dirName + pathSeparator + FileManager::getProjectFileName();
+        qDebug()<<projectFileName;
```
Comment (PeterLysyk) [inline src/mainwindow/mainwindow.cpp:1054]: remove

File: src/mainwindow/mainwindow.cpp [hunk 7]
```diff
@@ -1064,3 +1011,44 @@ void MainWindow::databaseDisconnect()
...
+    readWriteFileContent(projectFileName, databaseFileName);
+}
+
+void MainWindow::hideDatabaseFile()
+{
+    QString projectFileName = dirName + pathSeparator + FileManager::getProjectFileName();
+    qDebug()<<projectFileName;
```
Comment (PeterLysyk) [inline src/mainwindow/mainwindow.cpp:1061]: remove

File: src/mainwindow/mainwindow.cpp [hunk 8]
```diff
@@ -1064,3 +1011,44 @@ void MainWindow::databaseDisconnect()
...
+}
+
+void MainWindow::setDatabaseFileName()
+{
+    databaseFileName = dirName;
+    int position = dirName.lastIndexOf(QChar{pathSeparator});
+    databaseFileName += pathSeparator;
+    databaseFileName += dirName.mid(position + 1);
+    databaseFileName += projectDatabaseExtension;
+    qDebug()<<databaseFileName;
```
Comment (PeterLysyk) [inline src/mainwindow/mainwindow.cpp:1073]: remove

File: src/mainwindow/mainwindow.cpp [hunk 9]
```diff
@@ -1064,3 +1011,44 @@ void MainWindow::databaseDisconnect()
...
+    QString projectFileName = dirName + pathSeparator + FileManager::getProjectFileName();
+    qDebug()<<projectFileName;
+    readWriteFileContent(databaseFileName, projectFileName);
+    QFile::remove(databaseFileName);
+}
+
+void MainWindow::setDatabaseFileName()
+{
+    databaseFileName = dirName;
+    int position = dirName.lastIndexOf(QChar{pathSeparator});
```
Comment (PeterLysyk) [inline src/mainwindow/mainwindow.cpp:1069]: better use auto. compiler can consider, for example, unsigned as more suitable

File: src/mainwindow/mainwindow.cpp [hunk 10]
```diff
@@ -1064,3 +1011,44 @@ void MainWindow::databaseDisconnect()
...
+{
+    QFile fileToRead(fileToReadName);
+    QFile fileToWrite(fileToWriteName);
+
+    fileToRead.open(QIODevice::ReadOnly);
+    fileToWrite.open(QIODevice::WriteOnly);
+
+    QByteArray data = fileToRead.readAll();
+    fileToWrite.write(data);
+    qDebug()<<data;
```
Comment (PeterLysyk) [inline src/mainwindow/mainwindow.cpp:1086]: remove

File: src/mainwindow/mainwindow.h [hunk 11]
```diff
@@ -36,20 +36,19 @@ class MainWindow : public QMainWindow
...
 
-    bool mIsFinished;
 private:
     LocalConnectorInterface *mplocalConnector;
     Ui::MainWindow *ui;
     ProjectViewerDock *mpProjectViewerDock;
     ChatWindowDock *mpChatWindowDock;
     BottomPanelDock *mpBottomPanelDock;
-    QScopedPointer<DocumentManager> mpDocumentManager;    
+    QScopedPointer<DocumentManager> mpDocumentManager;
```
Comment (PeterLysyk) [inline src/mainwindow/mainwindow.h:46]: mDocumentManager. it's not a pointer

File: src/mainwindow/mainwindow.h [hunk 12]
```diff
@@ -36,20 +36,19 @@ class MainWindow : public QMainWindow
...
     Browser *mDocumentationBrowser;
```
Comment (PeterLysyk) [inline src/mainwindow/mainwindow.h:47]: I changed it in my request. See it and approve)

File: src/mainwindow/mainwindow.h [hunk 13]
```diff
@@ -64,8 +63,12 @@ class MainWindow : public QMainWindow
     void setDocumentFontFamily(const QString &fontFamily);
     void setDocumentFontSize(const QString &fontSize);
 
-    void databaseConnect(QString directory);
+    void databaseConnect();
     void databaseDisconnect();
+    void restoreDatabaseFile();
+    void hideDatabaseFile();
+    void setDatabaseFileName();
+    void readWriteFileContent(QString fileToReadName, QString fileToWriteName);
```
Comment (PeterLysyk) [inline src/mainwindow/mainwindow.h:74]: why you don't use const QString & ?

**Review by PeterLysyk — APPROVED** [review]
please, consider the amendments

**Review by opastushkov — APPROVED** [review]

**Pull request merged**